sql/analyzer/resolve_orderby.go

```diff
@@ -75,8 +78,14 @@ func pushdownSort(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel R
 			return n, transform.SameTree, nil
 		}
 
-		// If there are no columns required by the order by available, then move the order by
-		// below its child.
+		// If all the missing expressions are aliased, swap in the alias name and don't move the sort node
+		expressionToAliasName := aliasedExpressionsInNode(sort.Child)
```

Comment:
can a HAVING, FILTER or LIMIT node sit between the SORT and Projector? i.e. do we need to inspect the scope bottom-up? or top down iterate until we find nil or a Projector?

Reply:
Great question. Seems like `LIMIT` should always be above the `SORT` node to get the right results, but you have a great point about `FILTER` and `HAVING`. I just added some more tests to trigger this case, fixed the code, and pushed a new version. Thanks for suggesting this. 🙏